libp2p/protocols/pubsub/pubsubpeer.nim

```diff
@@ -43,9 +43,34 @@ type
 
   RPCHandler* = proc(peer: PubSubPeer, msg: seq[RPCMsg]): Future[void] {.gcsafe.}
 
-proc hash*(p: PubSubPeer): Hash = 
+func hash*(p: PubSubPeer): Hash = 
   # int is either 32/64, so intptr basically, pubsubpeer is a ref
-  cast[int](p).hash
+  cast[pointer](p).hash
```

Comment:
`hash(addr p[])` might be better here

Reply:
fine but, wouldn't this defect if nil?

Reply:
not sure tbh - ok let's leave cast for now